inst/stan/functions/obs_lpmf.stan

```diff
@@ -21,14 +22,25 @@ real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
     tar_obs = imp_obs[g][t];
     // allocate report day effects
     rdlh = srdlh[rdlurd[t:(t + l - 1), g]];
+    // allocate share of cases with known reference date
+    tar_alpha = alpha[g][t];
     // combine expected final obs and date effects to get expected obs
     esnap += l;
-    exp_obs[ssnap:esnap] = expected_obs(
+    vector[l] exp_obs_all = expected_obs(
       tar_obs, ref_lh[1:l, dpmfs[i]], rdlh, ref_p
     );
+    // compute expected final obs with known and missing reference date
+    exp_obs[ssnap:esnap] = exp_obs_all * tar_alpha;
+    if(t+l-1>=1+dmax){
+      exp_obs_miss[g][max(1 + dmax, t):(t + l - 1)] += exp_obs_all[max(2 + dmax - t, 1):l] * (1 - tar_alpha);
+    }
     ssnap += l;
   }
-  // observation error model (across all reference times and groups)
-  tar = neg_binomial_2_lupmf(snap_obs | exp_obs, phi);
+  // observation error model with known reference dates (across all reference times and groups)
+  tar = neg_binomial_2_lupmf(obs | exp_obs, phi);
+  // observation error model with missing reference dates
+  for (k in 1:n_groups) {
+    tar += neg_binomial_2_lupmf(obs_miss[k][(1 + dmax):n_obs_miss] | exp_obs_miss[k][(1 + dmax):n_obs_miss], phi);
```

Comment:
Currently, I am using exactly the same overdispersion as for the cases with known reference date. The question is whether we can think of typical scenarios in which the missingness has implications for the overdispersion. Then I should add a separate phi parameter for the `obs_miss`.

Reply:
At the moment there are no checks anywhere that `dmax` is not greater than the number of observations - so as implemented this is quite risky for going beyond the vector range of `obs_miss`.

Reply:
> dmax is not greater than the number of observations

This is a good point. Is this something to enforce in the R code or to allow and then build resilience to here (i.e with a min).

> , I am using exactly the same overdispersion as for the cases with known reference date

I can see the argument either way. For simplicity at least for now I vote we go with a single overdispersion parameter and circle back to this as and when.

Reply:
@seabbs Yes, I stated this limitation with `dmax` also further below... The question is whether fitting the model on a time window shorter than the maximum delay is a sensible idea at all / whether it would work. With a parametric delay model - maybe... If you think it would be wortwhile to support this (e.g. beginning of new outbreak), then we can of course add a case distinction/min